Pinta.Core/Classes/Document.cs

```diff
@@ -211,8 +211,7 @@ public void FinishSelection ()
 		Layer layer = Layers.SelectionLayer;
 
 		var g = new Cairo.Context (Layers.CurrentUserLayer.Surface);
-		selection.Clip (g);
-		layer.Draw (g);
+		layer.DrawWithoutOpacityProcessing (g, selection);
```

Comment:
I think it might also work to use `DrawWithOperator()` with the `Cairo.Operator.Source` operator, but please test!
That would avoid having multiple draw passes since it just replaces whatever pixels are within the clip area.

Reply:
Okay, yeah, I'm not at all familiar with Cairo and didn't know how to find exactly that operation. I'll give it a go :-)

Reply:
Hmm, when I use `DrawWithOperator` it completely replaces the target layer, erasing everything that isn't in the selection area. I tried calling `AppendPath(SelectionPath)` but it still does that. I don't see anything else on `Context` dealing paths, not sure what else to try.

Reply:
I'd also try restoring the original selection.Clip() call beforehand as well, if you weren't already

Reply:
I'll give that a go :-) Thanks

Reply:
It worked :-)